wearipedia/devices/garmin/fenix_gen.py

```diff
@@ -12,409 +13,2833 @@
+                "startDate": calendar_date,
+                "endDate": calendar_date,
+                "dateWeightList": [],
+                "totalAverage": {
+                    "from": int(date.timestamp() * 1000),
+                    "until": int((date + timedelta(days=1)).timestamp() * 1000),
+                    "weight": None,
+                    "bmi": None,
+                    "bodyFat": None,
+                    "bodyWater": None,
+                    "boneMass": None,
+                    "muscleMass": None,
+                    "physiqueRating": None,
+                    "visceralFat": None,
+                    "metabolicAge": None,
+                },
+            },
+        }
+
+        stats_and_body_aggregated_data.append(stats_and_body_aggregated_entry)
+
+    return stats_and_body_aggregated_data
+
+
+def get_heart_rate_data(start_date, num_days):
+    """Generate synthetic heart rate data for a specified date range.
+
+    This function generates synthetic heart rate data for a given date range, including various heart rate metrics
+    such as resting heart rate, maximum heart rate, minimum heart rate, and additional descriptors and values.
+
+    :param start_date: The start date in the format "YYYY-MM-DD".
+    :type start_date: str
+    :param num_days: The number of days for which to generate heart rate data.
+    :type num_days: int
+    :return: A list of dictionaries, each containing heart rate data for a specific day, including resting heart rate,
+        maximum and minimum heart rate, and additional heart rate descriptors and values.
+    :rtype: List[Dict]
+    """
+    heart_rate_data = []
+
+    for _ in range(num_days):
+        calendar_date = (
+            datetime.strptime(start_date, "%Y-%m-%d") + timedelta(days=_)
+        ).strftime("%Y-%m-%d")
+        start_timestamp_gmt = f"{calendar_date}T07:00:00.0"
+        end_timestamp_gmt = (
+            datetime.strptime(start_timestamp_gmt, "%Y-%m-%dT%H:%M:%S.0")
+            + timedelta(days=1)
+        ).strftime("%Y-%m-%dT%H:%M:%S.0")
+        start_timestamp_local = f"{calendar_date}T00:00:00.0"
+        end_timestamp_local = (
+            datetime.strptime(start_timestamp_local, "%Y-%m-%dT%H:%M:%S.0")
+            + timedelta(days=1)
+        ).strftime("%Y-%m-%dT%H:%M:%S.0")
+        resting_heart_rate = random.randint(50, 90)
+        max_heart_rate = random.randint(resting_heart_rate + 5, 120)
+        min_heart_rate = random.randint(resting_heart_rate + 2, max_heart_rate - 1)
+
+        last_seven_days_avg = random.randint(
+            resting_heart_rate - 2, resting_heart_rate + 2
+        )
+
+        heart_rate_values = []
+        start_datetime = datetime.strptime(start_timestamp_gmt, "%Y-%m-%dT%H:%M:%S.0")
+        end_datetime = datetime.strptime(end_timestamp_gmt, "%Y-%m-%dT%H:%M:%S.0")
+        while start_datetime < end_datetime:
+            heart_rate = random.randint(min_heart_rate, max_heart_rate)
```

Comment:
It seems that the heart rate data here is pretty noisy as it is being generated with simple rng. An improvement would be to use a more complex state generator. For an example, feel free to see the old garmin fenix 7s or polar h10/polar verity sense device generators.

Reply:
Took some time but Done!